.github/workflows/apidoc.yml

```
@@ -0,0 +1,151 @@
+    shell: /bin/bash -o errexit -o pipefail -o xtrace {0}
+
+jobs:
+  main:
+    runs-on: ubuntu-20.04
+
+    steps:
+      # Create a cache invalidation key based on the current year + week.
+      # This way, packages will be checked for updates once every week.
+      - name: Get Date
+        id: get-date
+        run: echo "::set-output name=date::$(/bin/date -u "+%Y%W")"
+
+      - name: Cache apt packages
+        id: cache-apt
+        uses: actions/cache@v2
+        with:
+          path: /var/cache/apt/archives
+          # The trailing number serves as a version flag that can be incremented
+          # to invalidate the cache after changing the list of packages.
+          key: ${{ github.workflow }}-${{ runner.os }}-${{ steps.get-date.outputs.date }}-apt-3
+
+      - name: Download apt packages
+        if: steps.cache-apt.outputs.cache-hit != 'true'
+        run: |
+          sudo apt-get update
+          sudo apt-get install --download-only -y --no-install-recommends \
+            asciidoctor \
+            cmake \
+            dbus-x11 \
+            gettext \
+            gir1.2-gtk-3.0 \
+            gir1.2-pango-1.0 \
+            git \
+            libdbus-1-dev \
+            libgirepository1.0-dev \
+            libnotify-bin \
+            libpango1.0-dev \
+            libstartup-notification0-dev \
+            libx11-xcb-dev \
+            libxcb-cursor-dev \
+            libxcb-icccm4-dev \
+            libxcb-keysyms1-dev \
+            libxcb-randr0-dev \
+            libxcb-shape0-dev \
+            libxcb-util0-dev \
+            libxcb-xfixes0-dev \
+            libxcb-xinerama0-dev \
+            libxcb-xkb-dev \
+            libxcb-xrm-dev \
+            libxcb-xtest0-dev \
+            libxdg-basedir-dev \
+            libxkbcommon-dev \
+            libxkbcommon-x11-dev \
+            xutils-dev \
+            liblua5.3-dev \
+            lua5.3
+
+      - name: Install downloaded packages
+        run: |
+          sudo dpkg -i /var/cache/apt/archives/*.deb
+
+      - name: Cache luarocks
+        id: cache-luarocks
+        uses: actions/cache@v2
+        with:
+          path: /tmp/luarocks
+          key: ${{ github.workflow }}-${{ runner.os }}-luarocks-3.5.0
+
+      - name: Install fresh Luarocks
+        if: steps.cache-luarocks.outputs.cache-hit != 'true'
+        run: |
+          wget -O /tmp/luarocks.tar.gz https://github.com/luarocks/luarocks/archive/v3.5.0.tar.gz
+          mkdir /tmp/luarocks
+          tar -xf /tmp/luarocks.tar.gz -C /tmp/luarocks --strip-components=1
+          cd /tmp/luarocks
+          ./configure
+          make build
+          sudo make install
+
+      - name: Install cached Luarocks
+        if: steps.cache-luarocks.outputs.cache-hit == 'true'
+        run: |
+          cd /tmp/luarocks
+          sudo make install
+
+      - name: Install rocks
+        run: |
+          sudo -H luarocks install lgi
+          sudo -H luarocks install ldoc
```

```suggestion
          sudo -H luarocks install lgi
          sudo -H luarocks install ldoc
          sudo -H luarocks install lua-discount
```

We should add `lua-discount` rock installation as it is what we use:

https://github.com/awesomeWM/awesome/blob/a4572b9b52d89369ce3bd462904d536ec116dc35/docs/config.ld#L11

(ldoc reference: https://stevedonovan.github.io/ldoc/manual/doc.md.html#Markdown_Support)

Not having this package installed results on the `format: using built-in markdown` message on your last "Update API doc" Actions run (https://github.com/sclu1034/awesome/runs/2371543797?check_suite_focus=true#step:14:421).

With this rock installed, it seems ldoc finishes faster on my machine.

---

I just figured there is a `lua-discount` package installed from apt-get in the "Build & test" workflow. (That explains why we don't have the "format message" in logs)

Maybe we should do the same here instead of installing the rock as Ubuntu seems to be more up-to-date than Luarocks (1.2.10.1 vs 2.1.8).

- https://packages.ubuntu.com/focal/lua-discount
- https://luarocks.org/modules/luarocks/lua-discount